coriolis/osmorphing/osmount/base.py

```diff
@@ -298,6 +299,99 @@ def _get_volume_block_devices(self):
         LOG.info("Volume block devices: %s", volume_devs)
         return volume_devs
 
+    def _find_dev_with_contents(self, devices, all_files=None,
+                                one_of_files=None):
+        if all_files and one_of_files:
+            raise exception.CoriolisException(
+                "all_files and one_of_files are mutually exclusive")
+        dev_name = None
+        for dev_path in devices:
+            dirs = None
+            tmp_dir = self._exec_cmd('mktemp -d').decode().split('\n')[0]
+            try:
+                self._exec_cmd('sudo mount %s %s' % (dev_path, tmp_dir))
+                # NOTE: it's possible that the device was mounted successfully
+                # but an I/O error occurs later along the line:
+                dirs = self._exec_cmd('ls %s' % tmp_dir).decode().split('\n')
+            except Exception:
+                self._event_manager.progress_update(
+                    "Failed to mount and scan device '%s'" % dev_path)
+                LOG.warn(
+                    "Failed to mount and scan device '%s':\n%s",
+                    dev_path, utils.get_exception_details())
+                utils.ignore_exceptions(self._exec_cmd)(
+                    "sudo umount %s" % tmp_dir
+                )
+                utils.ignore_exceptions(self._exec_cmd)(
+                    "sudo rmdir %s" % tmp_dir
+                )
+                continue
+
+            LOG.debug("Contents of device %s:\n%s", dev_path, dirs)
+
+            if all_files and dirs:
+                common = [i if i in dirs else None for i in all_files]
+                if not all(common):
+                    self._exec_cmd('sudo umount %s' % tmp_dir)
+                    continue
+
+                dev_name = dev_path
+                self._exec_cmd('sudo umount %s' % tmp_dir)
+            elif one_of_files and dirs:
+                common = [i for i in one_of_files if i in dirs]
+                if len(common) > 0:
+                    dev_name = dev_path
+                    self._exec_cmd('sudo umount %s' % tmp_dir)
+                    break
+            else:
+                self._exec_cmd('sudo umount %s' % tmp_dir)
+                continue
+
+        if dev_path:
+            devices.remove(dev_path)
+        return dev_name
+
+    def _find_and_mount_root(self, devices):
+        files = ["etc", "bin", "sbin", "boot"]
+        os_root_dir = None
+        os_root_device = self._find_dev_with_contents(
+            devices, all_files=files,
+        )
```

Comment:
NIT: these trailing parens 9both here and on 377+380) are quite an eyesore, could we just put them at the end of the previous line?

Reply:
Changed. The python plugin in my IDE automatically does this when I hit enter and the cursor is within the parenthesis.